src/web/js/dashboard/StudentList.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react';
+import Student from './Student.js';
+import { Button, Textfield, Card, CardTitle, CardText, Spinner } from 'react-mdl';
+
+class StudentList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      addingStudent: false,
+      newStudentFirstName: '',
+      newStudentLastName: '',
+      newStudentEmail: ''
+    };
+  }
+
+  handleClickAddStudent = () => {
+    this.setState({addingStudent: ! this.state.addingStudent});
+  }
+
+  handleChange = (event) => {
+    this.setState({[event.target.id]: event.target.value});
+  }
+
+  handleSubmitAddStudent = (event) => {
+    event.preventDefault();
+    this.props.snackBar('Student Added. Please allow a few seconds for changes to appear.');
+    this.setState({addingStudent: false});
+    this.props.api.addStudent({
+      firstName: this.state.newStudentFirstName.trim(),
+      lastName: this.state.newStudentLastName.trim(),
+      email: this.state.newStudentEmail.trim()
+    }).then((resp) => {
```

Comment:
So on all of the front-end calls that rely on a callback from the back-end/Google's APIs, I've added the following code:

```javascript
.then(resp => {
  //handle resp here
}).catch(e => {
  console.log(e);
  this.props.snackBar('Could not remove class (possibly network error). Try again in a few moments.');
});
```

(obviously with Snackbar text changed. This gets us three things:

- Some rudimentary error handling that will alert the user that something went wrong in the first place rather than silently failing.
- Something to look into should an error happen repeatedly. If someone wants to report an issue for us, we can have them look in the browser console, and the error from the backend will be logged there for us to investigate further.
- A basic framework for handling promise rejection on the front-end for future features.

I'll update this everywhere with appropriate messages.